src/renderer/services/file-management-system/index.ts

```diff
@@ -481,6 +484,16 @@ export default class FileManagementSystem {
     }
   }
 
+  public static chunksInFlightMaxForChunkSize(chunkSize: number){
+    for( let chunksInFlight = FileManagementSystem.CHUNKS_CEILING_IN_FLIGHT_REQUEST_CEILING_DEFAULT; chunksInFlight > 0; chunksInFlight-- ){
+      const bytesInFlight = chunksInFlight * chunkSize;
+      if(bytesInFlight <= FileManagementSystem.BYTES_IN_FLIGHT_CEILING){
+        return chunksInFlight;
+      }
+    }
+    return 1;//Should never reach here.
+  }
+
```

Comment:
Instead of this function, perhaps `Math.floor(BYTES_IN_FLIGHT_CEILING/chunkSize)`?

Reply:
wish I had though of that . .

Reply:
Though I think we still want to constrain it at 20 for small chunkSizes (for now at least).